### DISCUSSION TITLE
Running bazel server in docker container

### DISCUSSION THREAD

**Discussion by sonicxml** [bazelbuild/bazel#20202] [Q&A]
I would like to spin up a docker container that runs a long-running bazel server with the purpose of being able to send bazel build  commands to it (similar to a remote build server) from my machine or another container. What is the best / recommended way to do this? 

I saw the [bazel docker sandbox doc page](https://bazel.build/remote/sandbox) but it looks like it hasn't been updated in a while -- even the bazel-toolchains repo it links to doesn't have a sample `.bazelrc` past `4.1.0`. Is this still recommended?

I also tried playing around with the [bazel docker container](https://bazel.build/install/docker-container) but it looks like it's set up to be both client and server and not just run a server and listen for commands.

Any guidance would be appreciated, thanks!

**Comment by veith4f** [comment]
Great question. I am still relatively new to bazel but definitely wonder if in the heavily containerized world we have today there is still so much sense for platform-specific toolchains and rule implementations that ship different binaries for different platforms.

I just took over maintenance for a large bazel build env that going forward should be usable on different build machines comprising windows, linux and macos/amd64, macos/arm. 

Quite frankly, I think it is kind of ridiculous to implement treatments for all these different platforms. Bazel should run in a container where the respective bazel version is installed and bazel rules should make heavy use of bash and other gnu tools that the container provides. 

There should be a `bash` rule that allows running bash scripts and a large lib of rules that build on that `bash` rule to implement stuff like `base64-encode/decode`, `yq`, `jq`, `sed`, and all these other targets that sort of naturally suggest themselves.

I really wonder why that doesn't exist yet. Wouldn't it make much more sense to use bazel that way?

**Comment by veith4f** [comment]
Proposal: https://github.com/bazelbuild/bazel/discussions/26891
